src/Invoice/Domain/BackendInvoice/LnbitsBackendInvoice.php

```diff
@@ -41,21 +34,19 @@ public function requestInvoice(float $satsAmount, string $metadata): array
             // 'description_hash' => hash('sha256', $metadata),
         ];
 
-        $response = $this->httpFacade->post($endpoint, [
-            'headers' => [
+        $response = $this->httpApi->postRequestInvoice(
```

Comment:
Maybe in a future iteration, we can inject `options` directly to the `HttpClient` class, and we don't need to pass the `$endpoint` as a parameter, what do you think?

Something like:
```php
$response = $this->httpApi->postRequestInvoice(
    body: json_encode($content, JSON_THROW_ON_ERROR),
    headers: [
        'Content-Type' => 'application/json',
        'X-Api-Key' => $this->options['api_key'],
    ],
);
```

Reply:
Oh, nice observation. We can try that out in a follow up iteration. What are your thoughts on this, @Bashy ?

Reply:
It can be done as long as we don't have any other backend type.
Other kind of LN backend will have different `$endpoint`, `headers` and `body` values

Reply:
oh, that's right. Ok, let's keep it like it is for now. Thanks for bringing this idea up 😃

Reply:
If we are not going to have so many endpoints, my suggestion is to try to have specific methods in the `HttpApi` for each endpoint, where each of them can "auto-solve its own dependencies" (I mean, the parameters, endpoint, etc).

It's about concreteness over abstractness in this case, but to be honest, I am still not sure about the scalability of it, it would be nice to think over it in the future 😄